modules/js-and-css/optimize-oembeds/hooks.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Hook callbacks used for oEmbed Optimizer.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$oembed_lazy_load_scripts = false;
+
+/**
+ * Filter the oEmbed HTML.
+ *
+ * Add loading="lazy" to any iframe tags.
+ * Lazy load any script tags.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $html The oEmbed HTML.
+ * @return string
+ */
+function perflab_optimize_oembed_html( $html ) {
+	global $oembed_lazy_load_scripts;
+
+	// Find iframes and script tags in the oEmbed using the WP_HTML_Tag_Processor.
+	$p = new WP_HTML_Tag_Processor( $html );
+	if ( $p->next_tag( 'iframe' ) ) {
+		if ( empty( $p->get_attribute( 'loading' ) ) ) {
+			$p->set_attribute( 'loading', 'lazy' );
+			$html = $p->get_updated_html();
+		}
+	}
+	$p = new WP_HTML_Tag_Processor( $html );
+	if ( $p->next_tag( 'script' ) ) {
+		$oembed_lazy_load_scripts = true;
+		$p->set_attribute( 'data-lazy-embed-src', $p->get_attribute( 'src' ) );
+		$p->set_attribute( 'src', '' );
+		$html = $p->get_updated_html();
+	}
+
+	return $html;
```

Comment:
Seems these could be combined.

```suggestion
		}
	}
	$p = new WP_HTML_Tag_Processor( $html );
	if ( $p->next_tag( 'script' ) ) {
		$oembed_lazy_load_scripts = true;
		$p->set_attribute( 'data-lazy-embed-src', $p->get_attribute( 'src' ) );
		$p->set_attribute( 'src', '' );
	}

	return $p->get_updated_html();
```

Reply:
Sorry, I missed that there were two instances of `WP_HTML_Tag_Processor`. This is made obsolete anyway by https://github.com/WordPress/performance/pull/886#discussion_r1406860617

Reply:
good point. I'm going to combine these into a single loop anyway as @dmsnell suggested.